operator/internal/webhook/v1alpha1/konflux_webhook.go

```diff
@@ -0,0 +1,132 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	konfluxv1alpha1 "github.com/konflux-ci/konflux-ci/operator/api/v1alpha1"
+)
+
+// nolint:unused
+// log is for logging in this package.
+var konfluxlog = logf.Log.WithName("konflux-resource")
+
+// SetupKonfluxWebhookWithManager registers the webhook for Konflux in the manager.
+func SetupKonfluxWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).For(&konfluxv1alpha1.Konflux{}).
+		WithValidator(&KonfluxCustomValidator{Client: mgr.GetClient()}).
+		Complete()
+}
+
+// TODO(user): EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+
+// TODO(user): change verbs to "verbs=create;update;delete" if you want to enable deletion validation.
+// NOTE: The 'path' attribute must follow a specific pattern and should not be modified directly here.
+// Modifying the path for an invalid path can cause API server errors; failing to locate the webhook.
+// +kubebuilder:webhook:path=/validate-konflux-konflux-ci-dev-v1alpha1-konflux,mutating=false,failurePolicy=fail,sideEffects=None,groups=konflux.konflux-ci.dev,resources=konfluxes,verbs=create;update,versions=v1alpha1,name=vkonflux-v1alpha1.kb.io,admissionReviewVersions=v1
+
+// KonfluxCustomValidator struct is responsible for validating the Konflux resource
+// when it is created, updated, or deleted.
+//
+// NOTE: The +kubebuilder:object:generate=false marker prevents controller-gen from generating DeepCopy methods,
+// as this struct is used only for temporary operations and does not need to be deeply copied.
+type KonfluxCustomValidator struct {
+	Client client.Client
+}
+
+var _ webhook.CustomValidator = &KonfluxCustomValidator{}
+
+// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type Konflux.
+func (v *KonfluxCustomValidator) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	konflux, ok := obj.(*konfluxv1alpha1.Konflux)
+	if !ok {
+		return nil, fmt.Errorf("expected a Konflux object but got %T", obj)
+	}
+	konfluxlog.Info("Validation for Konflux upon creation", "name", konflux.GetName())
+
+	// Enforce singleton: only one Konflux instance is allowed per cluster
+	// List all existing Konflux instances and reject creation if any exist
+	// NOTE: This check has a race condition - if two ValidateCreate calls happen concurrently,
+	// both might pass this check before either instance is persisted. ValidateUpdate provides
+	// a secondary safety net to catch violations that slip through.
+	existingList := &konfluxv1alpha1.KonfluxList{}
```

Comment:
maybe include this code in a function, since we may have more validation code in the future.